tests/integration-e2e/test_reports.py

```diff
@@ -14,70 +14,50 @@ def _read_sonata_report(report_file):
-
-    # Modify the necessary fields
-    config["reports"] = config.get("reports", {})
-    config["reports"]["summation_report"] = {
-        "type": "summation",
-        "cells": "Mosaic",
-        "variable_name": "i_membrane,IClamp",
-        "sections": "all",
-        "dt": 0.1,
-        "start_time": 0.0,
-        "end_time": 40.0
+@pytest.mark.parametrize("create_tmp_simulation_config_file", [
+    {
+        "simconfig_fixture": "v5_sonata_config",
+        "extra_config": {
+            "node_set": "Mosaic",
+            "reports": {
+                "summation_report": {
+                    "type": "summation",
+                    "cells": "Mosaic",
+                    "variable_name": "i_membrane,IClamp",
+                    "sections": "all",
+                    "dt": 0.1,
+                    "start_time": 0.0,
+                    "end_time": 40.0
+                },
+                "synapse_report" : {
+                    "type": "synapse",
+                    "cells": "Mosaic",
+                    "variable_name": "ProbAMPANMDA_EMS.g",
+                    "sections": "all",
+                    "dt": 0.1,
+                    "start_time": 0.0,
+                    "end_time": 40.0
+                },
+                "summation_ProbGABAAB": {
+                    "type": "summation",
+                    "cells": "Mosaic",
+                    "variable_name": "ProbGABAAB_EMS.i",
+                    "sections": "all",
+                    "dt": 0.1,
+                    "start_time": 0.0,
+                    "end_time": 40.0
+                }
+            }
+        }
     }
-    config["reports"]["synapse_report"] = {
-        "type": "synapse",
-        "cells": "Mosaic",
-        "variable_name": "ProbAMPANMDA_EMS.g",
-        "sections": "all",
-        "dt": 0.1,
-        "start_time": 0.0,
-        "end_time": 40.0
-    }
-    # Added to verify no exception is raised when point process is not present in a section
-    config["reports"]["summation_ProbGABAAB"] = {
-        "type": "summation",
-        "cells": "Mosaic",
-        "variable_name": "ProbGABAAB_EMS.i",
-        "sections": "all",
-        "dt": 0.1,
-        "start_time": 0.0,
-        "end_time": 40.0
-    }
-
-    # Write the modified configuration to a temporary file in tmp_path
-    temp_config_path = tmp_path / "reports_config.json"
-    with open(temp_config_path, "w") as f:
-        json.dump(config, f, indent=4)
-
-    output_dir = Path(config["output"]["output_dir"])
-    if not output_dir.is_absolute():
-        output_dir = tmp_path / output_dir
-
-    return str(temp_config_path), str(output_dir)
-
-
+], indirect=True)
 @pytest.mark.slow
-def test_v5_sonata_reports(tmp_path):
+def test_v5_sonata_reports(create_tmp_simulation_config_file):
     import numpy.testing as npt
     from neurodamus import Neurodamus
+    from neurodamus.core.configuration import SimConfig
```

Comment:
As we agree before, the imports can be moved to the module level except the NeuronWrapper.

Reply:
ah I forgot about this